File: Workshop 3/src/BlackJack/model/rules/RulesFactory.java
/*
 * RulesFactory.java
 * Date: 26 Oktober, 2014
 * Author: Tobias Ohlsson (edited by Robert Willem Hallink)
 */
package BlackJack.model.rules;

/*
 * Mostly left untouched.
 * Added a new IWinStrategy implementation into this file.
 */
public class RulesFactory {
    
    public IHitStrategy GetHitRule()            { return new Soft17HitStrategy(); }
    public INewGameStrategy GetNewGameRule()    { return new AmericanNewGameStrategy(); }
    public IWinStrategy GetWinRule()            { return new EqualScoreDealerWinStrategy(); }
    
}

File: Workshop 3/Files/Peer Review/Materials - Eddy Proca/model/rules/RulesFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BlackJack.model.rules
{
    class RulesFactory
    {
        public IHitStrategy GetHitRule()
        {
            return new SoftHitStrategy();
            //return new BasicHitStrategy();
        }
        public INewGameStrategy GetNewGameRule()
        {
            return new AmericanNewGameStrategy();
            //return new InternationalNewGameStrategy();
        }
        public IWinRule GetWinRule()
        {
            return new DealerAdvantageWinRule();
            //return new PlayerAdvantageWinRule();
        }
        public IMaxScoreRule getMaxScoreRule() 
        {
            return new BlackJackMaxScoreRule();
        }
    }
}
